[Memoizee] Added string as a possible type to options/promise

benediktkaiser opened the pull request:
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

This PR modifies the types for [Memoizee](https://github.com/medikoo/memoizee). While recently using the types I noticed that the types for the [Memoizee Options](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/memoizee/index.d.ts#L14) is missing the string declaration for the Promise option. 

In their documentation one can see that promise can be `boolean`, `string` or `undefined`.
https://github.com/medikoo/memoizee#memoizing-asynchronous-functions (Can be found under supported modes).

This PR adds the possibility to use different promise modes to Memoizee.

juanpicado reviewed (approved):
LGTM, but I'd go further and add supported types for more accurate types
https://github.com/medikoo/memoizee#important-notice-on-internal-promises-handling

benediktkaiser commented:
> LGTM, but I'd go further and add supported types for more accurate types https://github.com/medikoo/memoizee#important-notice-on-internal-promises-handling

That sounds like a great idea. I completely forgot about doing that. Ill add the types now!

juanpicado commented:
Bot also suggest :) update tests, which also would be awesome.

benediktkaiser commented:
I have added the tests now. I was thinking of making a failing test at first (as one of the uses of 'done:finally' is to log thrown errors), but I noticed that would not work well in conjunction with the other tests. 

I added 3 more tests to the async function to see if the new types all work correctly.

juanpicado reviewed (approved):
LGTM v2.0 great stuff

juanpicado commented:
I love this friendly bot 💯

benediktkaiser commented:
Ready to merge

benediktkaiser commented:
Thank you @juanpicado for the quick reply and help! Have a nice day :D
